zerver/actions/streams.py

```diff
@@ -853,6 +854,27 @@ def do_change_stream_permission(
                 ).decode(),
             )
 
+    if old_invite_only_value and not stream.invite_only:
+        # We need to send stream creation event to users who can access the
+        # stream now but were not able to do so previously. So, we can exclude
+        # subscribers and realm admins from the non-guest user list.
+        stream_subscribers = private_stream_user_ids(stream.id)
+        old_can_access_stream_user_ids = stream_subscribers | {
+            user.id for user in stream.realm.get_admin_users_and_bots()
+        }
+        non_guest_user_ids = active_non_guest_user_ids(stream.realm_id)
```

Comment:
Did not directly use `can_access_stream_user_ids` here to get the users who can access the stream after changing, because it includes guest subscriptions too and will make an extra query for it but we do not need it since we anyway want to exclude subscribed users at the end.